### PR TITLE
Integrate GeoQuery into MapLoom

### PR DESCRIPTION
**Pull request by travislbrundage**
## What does this PR do?
Generalizes search and allows for integration of GeoQuery search endpoint.

### Screenshot
N/A (it just allows changing the endpoint in the search bar, nothing visible in UI)

### Related Issue
N/A

### CODE REVIEW

File: src/common/search/SearchService.js [hunk 1]
```diff
@@ -44,6 +44,54 @@
       return this;
     };
 
+    // algorithm taken from http://janmatuschek.de/LatitudeLongitudeBoundingCoordinates#Java
+    // from the "Boundingcoordinates" method
```
Comment (travislbrundage) [inline src/common/search/SearchService.js]: How do I appropriately credit this? I did not come up with this algorithm.

**Comment by theduckylittle** [comment]
@travislbrundage, most everything looks great, but I was wondering if you'd be willing to make one change:  Convert the manual URL and parameter construction to letting Angular handle it.

So instead of 
```
httpService_.get(url)
```

Use:
```
httpService_({
  method: 'GET',
  url: url,
  params: {
   q: '...',
   ...
 }
}).then( .... )
```

Whenever I go into fix or add something I try to use this format instead.  It helps avoid forgetting to escape a parameter during string construction and makes it easier to mixin new URL parameters later (specifically, when we need to add auth tokens to URL calls).

**Comment by travislbrundage** [comment]
Sure thing, I'll update and push again

**Comment by theduckylittle** [comment]
This is really close, can you remove the manual query string construction and use the "params" functionality provided by the HttpService?

Thanks!

**Comment by travislbrundage** [comment]
Oh, okay, my mistake, updating.

**Comment by theduckylittle** [comment]
Is this code working? There's a risk that having encodeURIComponent in the `params` object could double-encode the value part of the KVP in the Query string.  If it works then I can merge it.

Cheers.

**Comment by theduckylittle** [comment]
Looks great, thanks again!

**Pull request merged**